test/boost/server_stop_without_listening_test.cc

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include <chrono>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/with_timeout.hh>
+#include <seastar/testing/test_case.hh>
+#include <seastar/util/log.hh>
+
+#include "generic_server.hh"
+#include "utils/assert.hh"
+
+using namespace generic_server;
+using namespace logging;
+using namespace seastar;
+using namespace std::literals::chrono_literals;
+
+static logger test_logger("test_server");
+
+class test_server : public server {
+public:
+    test_server() : server("test_server", test_logger) {};
+protected:
+    [[noreturn]] shared_ptr<connection> make_connection(socket_address, connected_socket&&, socket_address) {
+        SCYLLA_ASSERT(false);
+    }
+};
+
+SEASTAR_TEST_CASE(server_stop_without_listening) {
+    test_server srv;
+    co_await with_timeout(lowres_clock::now() + 5min, srv.stop());
```

Comment:
If the test fails, shouldn't it be added after the fix? For the holy bisect.

Reply:
No, this order was intentional. For bisection, the tree must build at every stage, but it need not pass the test suite at every stage, IMO. I agree that actual bugs should not be introduced even temporarily (mid-series); however, this test case only exposes an existent bug, it does not introduce it.

The idea was that, in general, for a hypothesis to be any good, it has to be falsifiable; specifically, for this test to be worth adding, there should be a state of the tree where the test exists *and* fails -- so that the later bugfix can demonstrate through the test case that the fix makes a difference.

If you disagree, I can of course move the sorting order cleanup and the test case right after the fix (= conversion to `seastar::gate`).

Reply:
The test should indeed fail-before and pass-after, but sometimes I do bisect with the full ./test.py.

Reply:
So... should I move the test after the bugfix, or keep it at the front? :)

Reply:
After please

Reply:
done in v6